src/main/java/uk/org/tombolo/field/value/TimeseriesField.java

```diff
@@ -25,7 +27,11 @@ public TimeseriesField(String label, AttributeMatcher attribute) {
     @Override
     public JSONObject jsonValueForSubject(Subject subject, Boolean timeStamp) throws IncomputableFieldException {
         JSONArray arr = new JSONArray();
-        arr.addAll(TimedValueUtils.getBySubjectAndAttribute(subject, getAttribute()).stream().map(timedValue -> {
+        List<TimedValue> timedValues = TimedValueUtils.getBySubjectAndAttribute(subject, getAttribute());
+        if (timedValues.isEmpty()) {
+            throw new IncomputableFieldException(String.format("No TimedValue found for attribute %s", getAttribute().getLabel()));
+        }
+        arr.addAll(timedValues.stream().map(timedValue -> {
             JSONObject pair = new JSONObject();
```

Comment:
If it's empty we can just return after throwing the exception

Reply:
It won't get far after the exception is thrown. What I can do is create the json array object after so we don't waste one.